ArmyOfOnes/ArmyOfOnes/ViewControllers/CurrenciesListViewController.swift

```diff
@@ -0,0 +1,216 @@
+    
+    let quantityInput: UITextField = {
+        let textField = UITextField()
+        textField.text = String(1.0)
+        textField.placeholder = "Type a valid dollar quantity"
+        textField.layer.cornerRadius = 4.0
+        textField.layer.masksToBounds = true
+        textField.layer.borderColor = UIColor.lightGray.cgColor
+        textField.layer.borderWidth = 1.0
+        textField.textAlignment = .center
+        textField.keyboardType = UIKeyboardType.decimalPad
+        return textField
+    }()
+    
+    let cellIdentifier = "CellIdentifier"
+    let listTitle = "Army of Ones"
+    
+    var currencyRatesList = [String]()
+    var currencyCountries = [String]()
+    var currencyFormatter = NumberFormatter()
+    
+    var rates = [Currency]()
+    
+    //Mark: View methods
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        tableView.contentInset = UIEdgeInsetsMake(LayoutConstraints.tableTopInset.rawValue, 0, 0, 0);
+        tableView.isScrollEnabled = false
+        tableView.allowsSelection = false
+        tableView.separatorStyle = UITableViewCellSeparatorStyle.none
+        tableView.delegate = self
+        tableView.dataSource = self
+        
+        title = listTitle
+        tableView?.register(UITableViewCell.self, forCellReuseIdentifier: cellIdentifier)
+        
+        // Gesture that recognices tap to hide keyboard
+        let tapGesture = UITapGestureRecognizer(target: self, action:#selector(CurrenciesListViewController.endEditing(_:)))
+        tapGesture.cancelsTouchesInView = false
+        tableView.addGestureRecognizer(tapGesture);
+        
+        setQuantityInputLayout()
+        setupRates()
+        
+        view.setNeedsUpdateConstraints()
+    }
+    
+    func setQuantityInputLayout() -> Void {
+        
+        quantityInput.delegate = self
+        quantityInput.addTarget(self, action: #selector(CurrenciesListViewController.textFieldChanged(_:)), for: UIControlEvents.editingChanged)
+        navigationController?.view.insertSubview(quantityInput, belowSubview: navigationController!.navigationBar)
+        
+        //sets format number style to currency
+        currencyFormatter.numberStyle = NumberFormatter.Style.currency
+    }
+    
+    func showNoConnectionMessage() -> Void {
+        //shows an alert message when there is no internet connection
+        let alert = UIAlertController(title: "Connection error", message: "Make sure you are connected to the Internet", preferredStyle: UIAlertControllerStyle.alert)
+        alert.addAction(UIAlertAction(title: "OK", style: UIAlertActionStyle.default, handler: nil))
+        present(alert, animated: true, completion: nil)
+    }
+    
+    func setupRates() -> Void {
+        
+        let currencyController = CurrencyController()
+        currencyController.fetchRates(Country.allCountries) {(fetchedRates, error) -> Void in
+            
+            
+            guard let _ = error else  {
```

Comment:
Please check this [article](https://www.natashatherobot.com/swift-when-to-use-guard-vs-if/) about when to use `if` and when to use `guard`

Normally you use `guard` when you want to assert a certain variable exists before continuing with the normal flow of the application. In this case having an error is the exception and not the rule. I'd rather use `if let` in this specific case.